src/containers/Tenant/Diagnostics/Partitions/PartitionsControls/PartitionsControls.tsx

```diff
@@ -0,0 +1,170 @@
+    const [partitionIdSearchValue, setPartitionIdSearchValue] = useState('');
+
+    // Manual select control to enforce single-select behaviour for multiple select type
+    const [consumerSelectOpen, setConsumerSelectOpen] = useState(false);
+
+    useEffect(() => {
+        if (!partitions) {
+            return;
+        }
+
+        const partitionIdRe = new RegExp(escapeRegExp(partitionIdSearchValue), 'i');
+        const generalRe = new RegExp(escapeRegExp(generalSearchValue), 'i');
+
+        const filteredPartitions = partitions.filter((partition) => {
+            const {
+                partitionId,
+                readerName,
+                readSessionId,
+                partitionNodeId,
+                connectionNodeId,
+                partitionHost,
+                connectionHost,
+            } = partition;
+
+            const isPartitionIdMatch = partitionIdRe.test(partitionId);
+            const isOtherValuesMatch =
+                generalRe.test(readerName ?? '') ||
+                generalRe.test(readSessionId ?? '') ||
+                generalRe.test(String(partitionNodeId ?? '')) ||
+                generalRe.test(String(connectionNodeId ?? '')) ||
+                generalRe.test(partitionHost ?? '') ||
+                generalRe.test(connectionHost ?? '');
+
+            return isPartitionIdMatch && isOtherValuesMatch;
+        });
+
+        handleSearchChange(filteredPartitions);
+    }, [partitionIdSearchValue, generalSearchValue, partitions, handleSearchChange]);
+
+    const consumersToSelect = useMemo(
+        () =>
+            consumers
+                ? consumers.map((consumer) => ({
+                      value: consumer,
+                      content: consumer,
+                  }))
+                : undefined,
+        [consumers],
+    );
+
+    const columnsToSelect = useMemo(() => {
+        return initialColumnsIds.map((id) => {
+            return {
+                title: PARTITIONS_COLUMNS_TITILES[id as ValueOf<typeof PARTITIONS_COLUMNS_IDS>],
+                selected: Boolean(!hiddenColumns.includes(id)),
+                id: id,
+                required: id === PARTITIONS_COLUMNS_IDS.PARTITION_ID,
+            };
+        });
+    }, [initialColumnsIds, hiddenColumns]);
+
+    const handleConsumerSelectChange = (value: string[]) => {
+        // As we have selector with multiple options, the first value corresponds to previous value
+        // The second value is currently chosen consumer
+        handleSelectedConsumerChange(value[1]);
+        setConsumerSelectOpen(false);
+    };
+
+    const handlePartitionIdSearchChange = (value: string) => {
+        setPartitionIdSearchValue(value);
+    };
+
+    const handleGeneralSearchChange = (value: string) => {
+        setGeneralSearchValue(value);
+    };
+
+    const hadleTableColumnsSetupChange = (value: TableColumnSetupItem[]) => {
+        const result = [...hiddenColumns];
```

Comment:
Hidden columns are used now, because set of columns differs for partitions with consumers and partitions without consumers